### PR TITLE
Switch to monorepo

### PR DESCRIPTION
**Pull request by saleel**
- Switch to monorepo pattern with yarn workspace
- Move twitter specific circuit/contracts to separate package
- Add tests for email verifier circuit
- Add tests for twitter circuit
- Add helpers library with reusable utils

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -53,9 +53,9 @@ circuits/circuit*zkey*
 circuits/email/
 cache/
 test.log
-src/contracts/out/
-src/contracts/lib/
-src/contracts/broadcast
+pacakges/contracts/out/
```
Comment (Divide-By-0) [inline .gitignore:56]: typo on packages

File: dizkus-scripts/4_gen_vkey.sh
Comment (Divide-By-0) [inline dizkus-scripts/4_gen_vkey.sh:1]: wait so where did this file go? is there an updated readme for how to compile/run?

Comment (saleel) [reply dizkus-scripts/4_gen_vkey.sh:1]: The scripts folder was moved to `packages/twitter-verifier-circuits/scripts` along with the circom files for twitter.
I have to update the paths in Readme. I will do it if the stuctures makes sense.

**Comment by Divide-By-0** [comment]
Cool -- some minor merge conflicts but other than that, code looks fine to me. Would be great to update the readme so its up to date with the new structure! 

I don't have enough experience with npm package structures so willing to trust your experience here on that.

**Comment by saleel** [comment]
Yea, I will fix the conflicts and update the Readme. Some existing stuff in the Readme was also updated if I remember correctly.
Structure wise this is good IMO. The main change was splitting the Twitter circuit in to two - new tests work well but wanted to be sure I have not missed anything.

**Pull request merged**